docs/source/cuda/examples.rst

```diff
@@ -52,35 +60,123 @@ memory::
-                tmp += sA[tx, j] * sB[j, ty]
+                tmp += sA[ty, j] * sB[j, tx]
 
             # Wait until all threads finish computing
             cuda.syncthreads()
+        if y < C.shape[0] and x < C.shape[1]:
+            C[y, x] = tmp
 
-        C[x, y] = tmp
-
-Because the shared memory is a limited resources, the code preloads small
+Because the shared memory is a limited resource, the code preloads small
 block at a time from the input arrays.  Then, it calls
 :func:`~numba.cuda.syncthreads` to wait until all threads have finished
 preloading and before doing the computation on the shared memory.
 It synchronizes again after the computation to ensure all threads
 have finished with the data in shared memory before overwriting it
 in the next loop iteration.
 
+An example usage of this function is as follows:
+
+.. code-block:: python
+  :linenos:
+
+    x_h = np.arange(16).reshape([4,4])
+    y_h = np.ones([4,4])
+    z_h = np.zeros([4,4])
+
+    x_d = cuda.to_device(x_h)
+    y_d = cuda.to_device(y_h)
+    z_d = cuda.to_device(z_h)
+
+    TPB = 3
+    threadsperblock = (TPB, TPB)
+    blockspergrid_x = math.ceil(z_h.shape[0] / threadsperblock[0])
+    blockspergrid_y = math.ceil(z_h.shape[1] / threadsperblock[1])
+    blockspergrid = (blockspergrid_x, blockspergrid_y)
+
+    fast_matmul[blockspergrid, threadsperblock](x_d, y_d, z_d)
+    z_h = z_d.copy_to_host()
+    print(z_h)
+    print(x_h@y_h)
+
+This passes a :ref:`CUDA memory check test <debugging-cuda-python-code>`:
+
+.. code-block:: none
+
+    $ cuda-memcheck python t49.py
+    ========= CUDA-MEMCHECK
+    [[ 6.  6.  6.  6.]
+    [22. 22. 22. 22.]
+    [38. 38. 38. 38.]
+    [54. 54. 54. 54.]]
+    [[ 6.  6.  6.  6.]
+    [22. 22. 22. 22.]
+    [38. 38. 38. 38.]
+    [54. 54. 54. 54.]]
+    ========= ERROR SUMMARY: 0 errors
+
+Note: For high performance matrix multiplication operations in CUDA, see the `cuBLAS API from pyculib <http://pyculib.readthedocs.io/en/latest/cublas.html#pyculib.blas.Blas.gemm>`_ or the `CuPy implementation <https://docs.cupy.dev/en/stable/reference/generated/cupy.matmul.html>`_.
+
+The approach outlined here generalizes to non-square matrix multiplication as follows by adjusting the blockspergrid variable:
+
+Again, here is an example usage:
+
+.. code-block:: python
+  :linenos:
+
+  x_h = np.arange(115).reshape([5,23])
+  y_h = np.ones([23,7])
+  z_h = np.zeros([5,7])
+
+  x_d = cuda.to_device(x_h)
+  y_d = cuda.to_device(y_h)
+  z_d = cuda.to_device(z_h)
+
+  #TPB must be an integer between 1 and 32
+  TPB = 32
+  threadsperblock = (TPB, TPB)
+  grid_y_max = max(x_h.shape[0],y_h.shape[0])
+  grid_x_max = max(x_h.shape[1],y_h.shape[1])
+  blockspergrid_x = math.ceil(grid_x_max / threadsperblock[0])
+  blockspergrid_y = math.ceil(grid_y_max / threadsperblock[1])
+  blockspergrid = (blockspergrid_x, blockspergrid_y)
+
+  fast_matmul[blockspergrid, threadsperblock](x_d, y_d, z_d)
+  z_h = z_d.copy_to_host()
+  print(z_h)
+  print(x_h@y_h)
```

Comment:
Omitting whitespace between the infix matrix multiplication operator and its operands seems to confuse Sphinx (or at least the version I'm using, anyway, 4.0.1) and it emboldens the second operand. Adding some whitespace seems to fix this:

```suggestion
  print(x_h @ y_h)
```

(I note also that adding whitespace seems to be the idiomatic thing to do, as it's in the examples in PEP 465: https://www.python.org/dev/peps/pep-0465/)